service.subtitles.supersubtitles/service.py

```diff
@@ -0,0 +1,577 @@
+
+    searchlist.sort(key=lambda x: (x['score'], x['language_eng'] == item['preferredlanguage'], x['language_eng'],
+                                   releaser.lower() in x['filename'].lower() if releaser else x['filename']),
+                    reverse=True)
+
+    return searchlist
+
+
+def search_subtitles(item, recursive=True):
+    if not item['season'] and not item['episode']:
+        debuglog("No season or episode info found for %s" % item['tvshow'])
+        return None
+
+    showids = get_showids(item)
+    if not showids:
+        debuglog("No ids found for %s" % item['tvshow'])
+        if recursive and normalize_string(xbmc.getInfoLabel("VideoPlayer.TVshowtitle")):
+            debuglog("Second try: search by filename")
+            return search_subtitles(setup_tvshow_data(item, False), False)
+        else:
+            return None
+
+    searchdict = {}
+    for showid in showids:
+        subtitles = search_subtitles_for_show(item, showid)
+        if subtitles:
+            avg = sum(x['score'] for x in subtitles) / float(len(subtitles))
+            searchdict[(avg, showid)] = subtitles
+
+    searchlist = []
+    for key in sorted(searchdict, reverse=True):
+        searchlist.extend(searchdict[key])
+
+    return searchlist
+
+
+def search(item):
+    debuglog(item)
+    subtitles_list = search_subtitles(item)
+
+    if subtitles_list:
+        index = 0
+        for it in subtitles_list:
+            index += 1
+            #label="%s | %s | %s"%(it['name'], it['filename'], it['uploader'])
+            label = "%s [%s]" % (it['filename'], it['uploader'])
+
+            if it['seasonpack']:
+                label += (' (%s)' % (__language__(32503)))
+
+            listitem = xbmcgui.ListItem(label=it['language_eng'],
+                                        label2=label,
+                                        iconImage=it['rating'],
+                                        thumbnailImage=it['flag']
+            )
+            listitem.setProperty('sync', ('false', 'true')[it['sync']])
+            listitem.setProperty('hearing_imp', ('false', 'true')[it.get('hearing', False)])
+
+            qparams = {'action': 'download', 'actionsortorder': str(index).zfill(2), 'id': it['id'],
+                       'filename': py2_encode(it['filename'], 'utf-8')}
+            url = "plugin://%s/?%s" % (__scriptid__, urlencode(qparams))
+
+            xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=listitem, isFolder=False)
+
+
+def is_archive(filename):
+    if filename:
+        for ext in ARCHIVE_EXTENSIONS:
+            if filename.endswith(ext):
+                return True
+    return False
+
+
+def download_file(item):
+    filename = py2_decode((unquote_plus(py2_decode(item['filename'], "utf-8")).replace(' ', '_')), 'utf-8')
+    localfile = os.path.join(__temp__, filename)
+    qparams = {'action': 'letolt', 'felirat': item['id']}
+
+    response = send_request(qparams)
```

Comment:
Same story with the `with send_request(qparams) as reponse;` to make sure resources are released